torcx/metadata-and-systemd-target.md

```diff
@@ -0,0 +1,76 @@
+# Metadata and systemd target
+
+In many cases, it is desirable to inspect the state of a system booted with Torcx and to verify the details of the configuration that has been applied.
+For these purposes, Torcx comes with additional facilities to integrate with systemd-based workflows: a custom target and a metadata file containing environment flags.
+
+## Metadata entries and environment flags
+
+In order to signal a successful run, Torcx writes a metadata file at most once per boot. The format of this file is suitable for consumption by systemd `EnvironmentFile=` [directive][systemd-exec] and can be used to introspect the booted configuration at runtime.
+
+The metadata file is written to `/run/metadata/torcx` and contains a list of key-value pairs:
+
+```
+$ cat /run/metadata/torcx
+
+TORCX_LOWER_PROFILES="vendor"
+TORCX_UPPER_PROFILE="custom-demo"
+TORCX_PROFILE_PATH="/run/torcx/profile.json"
+TORCX_BINDIR="/run/torcx/bin"
+TORCX_UNPACKDIR="/run/torcx/unpack"
+```
+
+These values can be used to detect where assets have been unpacked and propagated (shown above as "unpack" and "bin" entries), which profiles have been sourced (both vendor- and user-provided), and what is the resulting profile that has been applied.
+
+Finally, the runtime profile can be inspected to detect which addons (and versions) are currently applied:
+
+```
+$ cat /run/torcx/profile.json
+
+{
+  "kind": "profile-manifest-v0",
+  "value": {
+    "images": []
+  }
+}
+```
+
+## Torcx target unit
+
+System services may depend on successful execution of Torcx generator. As such, `torcx.target` is provided as a target unit which is only reached if the generator run without failures and the system got successfully sealed.
```

Comment:
Reworded the sentence following @squeed's suggestion to avoid the empasse and turning it into a simpler form.